scenario/mocking.py

```diff
@@ -69,26 +70,51 @@ class ActionMissingFromContextError(Exception):
 
 
 class _MockExecProcess:
-    def __init__(self, command: Tuple[str, ...], change_id: int, out: "ExecOutput"):
+    def __init__(
+        self,
+        command: Tuple[str, ...],
+        change_id: int,
+        exec: "Exec",
+        container: "ContainerSpec",
+    ):
         self._command = command
         self._change_id = change_id
-        self._out = out
+        self._exec = exec
         self._waited = False
-        self.stdout = StringIO(self._out.stdout)
-        self.stderr = StringIO(self._out.stderr)
+        self.stdout = StringIO(self._exec.stdout)
+        self.stderr = StringIO(self._exec.stderr)
+        self.stdin = StringIO(self._exec.stdin)
```

Comment:
I hope that no charms use streaming stdin/stdout to pebble, like `pebble exec --tty telnet` or something 🙈 

Then again, I'd rather keep the scope small, even if one single forgotten charm does this; it's a new library / new major version of the library.

Reply:
`Harness` has support for a fuller range of the actual Pebble.exec functionality (the docs have details). For Scenario, at least in 7.0, the idea is to keep it smaller and extend it as necessary, but trying to ensure that we have done all the breaking changes at this point.